install-devstack-xen.sh

```diff
@@ -415,100 +428,31 @@ cd devstack*
 set -exu
 
 cd $TMPDIR
 
 cd devstack*
 
-cat << LOCALCONF_CONTENT_ENDS_HERE > local.conf
-# ``local.conf`` is a user-maintained settings file that is sourced from ``stackrc``.
-# This gives it the ability to override any variables set in ``stackrc``.
-# The ``localrc`` section replaces the old ``localrc`` configuration file.
-# Note that if ``localrc`` is present it will be used in favor of this section.
-# --------------------------------
-[[local|localrc]]
-
-enable_plugin os-xenapi https://github.com/openstack/os-xenapi.git
-
-# Passwords
-MYSQL_PASSWORD=citrix
-SERVICE_TOKEN=citrix
-ADMIN_PASSWORD=citrix
-SERVICE_PASSWORD=citrix
-RABBIT_PASSWORD=citrix
-GUEST_PASSWORD=citrix
-XENAPI_PASSWORD="$XENSERVER_PASS"
-SWIFT_HASH="66a3d6b56c1f479c8b4e70ab5c2000f5"
-
-# Nice short names, so we could export an XVA
-VM_BRIDGE_OR_NET_NAME="osvmnet"
-PUB_BRIDGE_OR_NET_NAME="ospubnet"
-XEN_INT_BRIDGE_OR_NET_NAME="osintnet"
-
-# Do not use secure delete
-CINDER_SECURE_DELETE=False
-
-# Compute settings
-VIRT_DRIVER=xenserver
-
-# OpenStack VM settings
-OSDOMU_VDI_GB=30
-OSDOMU_MEM_MB=8192
-
-TERMINATE_TIMEOUT=90
-BUILD_TIMEOUT=600
-
-# DevStack settings
-LOGFILE=${SCREEN_LOGDIR}/stack.log
-SCREEN_LOGDIR=${SCREEN_LOGDIR}
-
-UBUNTU_INST_HTTP_HOSTNAME=archive.ubuntu.com
-UBUNTU_INST_HTTP_DIRECTORY=/ubuntu
-
-# Turn on verbosity (password input does not work otherwise)
-VERBOSE=True
-
-# XenAPI specific
-XENAPI_CONNECTION_URL="http://$XENSERVER_IP"
-VNCSERVER_PROXYCLIENT_ADDRESS="$XENSERVER_IP"
-
-# Neutron specific part
-ENABLED_SERVICES+=neutron,q-domua
-Q_ML2_PLUGIN_MECHANISM_DRIVERS=openvswitch
-
-Q_ML2_PLUGIN_TYPE_DRIVERS=vlan,flat
-ENABLE_TENANT_TUNNELS=False
-ENABLE_TENANT_VLANS=True
-Q_ML2_TENANT_NETWORK_TYPE=vlan
-ML2_VLAN_RANGES="physnet1:1100:1200"
-
-PUB_IP=172.24.4.1
-SUBNETPOOL_PREFIX_V4=192.168.10.0/24
-NETWORK_GATEWAY=192.168.10.1
-
-VLAN_INTERFACE=eth1
-PUBLIC_INTERFACE=eth2
-
-# Nova user specific configuration
-# --------------------------------
-[[post-config|\\\$NOVA_CONF]]
-[DEFAULT]
-disk_allocation_ratio = 2.0
-
-# Neutron ovs bridge mapping
-[[post-config|\\\$NEUTRON_CORE_PLUGIN_CONF]]
-[ovs]
-bridge_mappings = physnet1:br-eth1,public:br-ex
-
-LOCALCONF_CONTENT_ENDS_HERE
-
-# unset nounset for $NOVA_CONF
-set -u
+# Configure local.conf
+wget https://raw.githubusercontent.com/citrix-openstack/qa/master/local.conf.sample
+cp local.conf.sample local.conf
+sed -i "s/GUEST_NAME=/GUEST_NAME=$NODE_NAME/g" local.conf
+sed -i "s/VNCSERVER_PROXYCLIENT_ADDRESS=/VNCSERVER_PROXYCLIENT_ADDRESS=$XENSERVER/g" local.conf
+sed -i "s/XENAPI_PASSWORD=/XENAPI_PASSWORD=$XENSERVER_PASS/g" local.conf
+sed -i "s/UBUNTU_INST_HTTP_HOSTNAME=archive.ubuntu.com/UBUNTU_INST_HTTP_HOSTNAME=$UBUNTU_REPO_URL/g" local.conf
+
+if [ "$NODE_TYPE" = "compute" ]; then
+    sed -i "s/ENABLED_SERVICES+=/#ENABLED_SERVICES+=/g" local.conf
+    sed -i "s/#ENABLED_SERVICES=/ENABLED_SERVICES=/g" local.conf
+    sed -i "s/#SERVICE_HOST/SERVICE_HOST/g" local.conf
```

Comment:
Why not insert these into local.conf rather than uncomment them?  Having them in local.conf commented is strange.